File: 2021/08/Part1.cpp
#include <string>
#include <vector>

#include "lib.h"

/**
 * Simply parse the input and count the number of strings after the '|' that
 * have length 2, 3, 4, or 7, representing numbers 1, 7, 4, and 8 respectively
 */
int p1(std::string filename) {
    int total = 0;
    for (std::string line: readLines(filename)) {
        std::vector<std::string> parts = split(line, " | ");
        std::vector<std::string> outputs = split(parts.at(1), " ");
        for (std::string output: outputs) {
            switch (output.length()) {
                case 2:
                case 3:
                case 4:
                case 7:
                    total++;
                    break;
            }
        }
    }
    return total;
}

int main() {
    assertEqual(p1("input_test.txt"), 26);
    int ans = p1("input.txt");
    printf("Ans = %d\n", ans);
    return 0;
}

File: 2022/10/Part1.py
cycle = 1
X = 1
ans = 0
with open("input.txt") as f:
    for line in f:
        line = line.strip()
        if cycle in [20, 60, 100, 140, 180, 220]:
            ans += cycle * X
        if line == "noop":
            cycle += 1
        elif line != "": # addx
            if cycle + 1 in [20, 60, 100, 140, 180, 220]:
                ans += (cycle + 1) * X
            V = int(line.split()[1])
            X += V
            cycle += 2

print(ans)